swift/internal/derived_files.bzl

```diff
@@ -17,84 +17,132 @@
 load("@bazel_skylib//lib:paths.bzl", "paths")
 load(":utils.bzl", "owner_relative_path")
 
-def _ast(actions, target_name, src):
+def _default_path(ctx, basename):
+    target_name = ctx.label.name
+    return paths.join(target_name, basename)
+
+def _declare_file(actions, target_name, basename, add_target_name_to_output_path):
+    if add_target_name_to_output_path:
+        return actions.declare_file(paths.join(target_name, basename))
+    else:
+        return actions.declare_file(basename)
+
+def _declare_directory(actions, target_name, directory, add_target_name_to_output_path):
+    if add_target_name_to_output_path:
+        return actions.declare_directory("{}_{}".format(target_name, directory))
+    else:
+        return actions.declare_directory(directory)
```

Comment:
Not sure if that works in this case, but if we are in control of  `actions`, we could wrap them on the call sites and move that logic to the wrapper itself. That could significantly reduce the number of method to which the flag is passed.

Reply:
@sewerynplazuk you mean [these](https://bazel.build/rules/lib/builtins/actions) actions? Could you give an example please? 🙏

Reply:
There's something like `__getattr__` in Python. Not sure if Starlark supports it.

Anyway, after a second though I think it'd be a bit shady change. So up to you.

Reply:
I think is alright as is, agree the proposed solution is a bit too much